### PR TITLE
test: Add `v1beta1` testing for the garbage collection controller

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

This PR adds `v1beta1` testing for the garbage collection controller for the `NodeClaim` resource

**How was this change tested?**

`make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/nodeclaim/garbagecollection/machine_test.go [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+				defer wg.Done()
+
+				_, err := cloudProvider.Get(ctx, fake.ProviderID(id))
+				Expect(err).To(HaveOccurred())
+				Expect(corecloudprovider.IsNodeClaimNotFoundError(err)).To(BeTrue())
+			}(id)
+		}
+		wg.Wait()
+	})
+	It("should not delete all instances if they all have machine owners", func() {
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply pkg/controllers/nodeclaim/garbagecollection/machine_test.go:168]: No, the one in the `suite_test.go` is testing what happens if both Machines and NodeClaims are combined. It just ensures that the two resources play nice together

File: pkg/controllers/nodeclaim/garbagecollection/nodeclaim_test.go [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+			ProviderID: providerID,
+		})
+		ExpectApplied(ctx, env.Client, nodeClaim, node)
+
+		ExpectReconcileSucceeded(ctx, garbageCollectionController, client.ObjectKey{})
+		_, err := cloudProvider.Get(ctx, providerID)
+		Expect(err).ToNot(HaveOccurred())
+		ExpectExists(ctx, env.Client, node)
+	})
+	It("should not delete many instances or nodes if they already have NodeClaim owners that match it", func() {
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply pkg/controllers/nodeclaim/garbagecollection/nodeclaim_test.go:290]: Same response to the question above

File: pkg/controllers/nodeclaim/garbagecollection/machine_test.go [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+		ExpectApplied(ctx, env.Client, node)
+
+		ExpectReconcileSucceeded(ctx, garbageCollectionController, client.ObjectKey{})
+		_, err := cloudProvider.Get(ctx, providerID)
+		Expect(err).To(HaveOccurred())
+		Expect(corecloudprovider.IsNodeClaimNotFoundError(err)).To(BeTrue())
+
+		ExpectNotFound(ctx, env.Client, node)
+	})
+	It("should delete many instances if they all don't have machine owners", func() {
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply pkg/controllers/nodeclaim/garbagecollection/machine_test.go:115]: No, the one in the suite_test.go is testing what happens if both Machines and NodeClaims are combined. It just ensures that the two resources play nice together

**Review by engedaam — APPROVED** [review]
LGTM!

**Pull request merged**